packages/x-data-grid/src/components/GridLoadingOverlay.tsx

```diff
@@ -1,13 +1,63 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
+import LinearProgress from '@mui/material/LinearProgress';
 import CircularProgress from '@mui/material/CircularProgress';
+import { Theme, SystemStyleObject } from '@mui/system';
 import { GridOverlay, GridOverlayProps } from './containers/GridOverlay';
+import { GridSkeletonLoadingOverlay } from './GridSkeletonLoadingOverlay';
+import { useGridApiContext } from '../hooks/utils/useGridApiContext';
+
+type GridLoadingOverlayVariant = 'circular-progress' | 'linear-progress' | 'skeleton';
+
+export interface GridLoadingOverlayProps extends GridOverlayProps {
+  /**
+   * The variant of the overlay.
+   * @default 'circular-progress'
+   */
+  variant?: GridLoadingOverlayVariant;
+  /**
+   * The variant of the overlay when no rows are displayed.
+   * @default 'circular-progress'
+   */
+  noRowsVariant?: GridLoadingOverlayVariant;
+}
+
+const LOADING_VARIANTS: Record<
+  GridLoadingOverlayVariant,
+  {
+    component: React.ComponentType;
+    sx: SystemStyleObject<Theme>;
+  }
+> = {
+  'circular-progress': {
+    component: CircularProgress,
+    sx: {},
+  },
+  'linear-progress': {
+    component: LinearProgress,
+    sx: { display: 'block' },
+  },
+  skeleton: {
+    component: GridSkeletonLoadingOverlay,
+    sx: {
+      display: 'block',
+      background: 'var(--DataGrid-containerBackground)',
```

Comment:
We require a solid background on the skeleton overlay to avoid this:
![localhost_3001_playground_ (2)](https://github.com/mui/mui-x/assets/9557798/6c588bc9-165b-49e3-b63c-81fa40420424)

I am wondering if the `containerBackground` is the wrong variable to use here, as you may want to have a different header/pinned background color to the actual data grid background color. On the latter point, we don't seem to be explicitly setting a background color on the data grid.

I am wondering if we should introduce another theme variable here for the background:
```tsx
MuiDataGrid: {
  // Body, footer and solid overlays
  background: '#ffffff',
  // Pinned columns sections
  pinnedBackground: '#340606',
  // Headers, and top & bottom fixed rows
  containerBackground: '#343434',
}
```

Reply:
There is already a variable for the default background color in the theme (not specific to datagrid), let's use that one until we have a request for another variable.

Reply:
Is it an option to hide the underlying UI? Currently it still seems to be a goal of the grid to adopt underlying background (even though headers and pinned columns don't follow this exactly)

Reply:
> Is it an option to hide the underlying UI instead?

This may be a safer option than using the default background color from the theme, as we cannot be sure that that is the background color applied on the body/element behind the data grid.

Reply:
Removed in https://github.com/mui/mui-x/pull/13293/commits/ceb9c7e279436afabf509ee34ba8872f19b94c60